src/identity.rs

```diff
@@ -221,13 +221,19 @@ impl Password {
         S2: Into<String>,
         S3: Into<String>,
     {
-        let url = auth_url.into_url()?;
-        // TODO: more robust logic?
-        let token_endpoint = if url.path().ends_with("/v3") {
+        let mut url = auth_url.into_url()?;
+
+        let _p = url
+            .path_segments_mut()
+            .map_err(|_| "cannot be base")?
```

Comment:
I'd prefer to avoid relying on an implicit conversion from a string to `Error`, let's use `Error::new(ErrorKind::InvalidConfig, "Invalid auth_url:  wrong schema?")` or something like that.

Reply:
oh cool, thanks!